File: gpt-3.5-ft-large/template-context-full/parallel_construct_num_gangs_clause.c
#include "acc_testsuite.h"
#ifndef T1
//T1:parallel construct num_gangs clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    if (acc_get_device_type() == acc_device_none){
        return err;
    }
    #pragma acc parallel num_gangs(2222, 1)
    {
        if (acc_get_device_type() == acc_device_host){
            if (acc_get_num_gangs(1) != acc_get_device_num_gangs(1)){
                err += 1;
            }
            if (acc_get_num_gangs(1)*2222 == acc_get_device_num_gangs(1)){
                err += 1;
            }
        }
        #pragma acc loop gang
        for (int x = 0; x < 10; ++x){
            ;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-ft-large/stage2/parallel_construct_num_gangs_clause.f90
#ifndef T1
!T1:parallel construct num_gangs clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        REAL(8),DIMENSION(10, 10) :: a, b, c
        INTEGER :: x, y
        
        SEEDDIM(1) = 1
#if REAL
        REAL SEEDS(5)
        SEEDDIM(0) = 5
#endif
#if DOUBLE
        DOUBLE PRECISION SEEDS(5)
        SEEDDIM(0) = 2
#endif
        CALL RANDOM_SEED(PUT=SEEDS)
        
        CALL RANDOM_NUMBER(a)
        CALL RANDOM_NUMBER(b)
        c = 0
        
        !$acc data copyin(a(1:10, 1:10), b(1:10, 1:10)) copyout(c(1:10, 1:10))
          !$acc parallel num_gangs(2, 5)
            !$acc loop gang
            DO y = 1, 2
              !$acc loop worker
              DO x = 1, 10
                c(x, y) = c(x, y) + a(x, y) + b(x, y)
              END DO
            END DO
          !$acc end parallel
        !$acc end data

        DO y = 1, 2
          DO x = 1, 10
            IF (abs(c(x, y) - (a(x, y) + b(x, y))) .gt. PRECISION) THEN
              errors = errors + 1
            END IF
          END DO
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM